packages/webview_flutter/webview_flutter_wkwebview/lib/src/web_kit_webview_widget.dart

```diff
@@ -161,6 +161,18 @@ class WebKitWebViewPlatformController extends WebViewPlatformController {
     };
   }
 
+  @override
+  Future<void> clearCache() {
+    return webView.configuration.webSiteDataStore.removeDataOfTypes(
+      <WKWebsiteDataTypes>{
+        WKWebsiteDataTypes.memoryCache,
+        WKWebsiteDataTypes.diskCache,
+        WKWebsiteDataTypes.offlineWebApplicationCache,
+      },
+      DateTime.fromMillisecondsSinceEpoch(0),
+    );
+  }
```

Comment:
This is different than the current implementation: https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter_wkwebview/ios/Classes/FlutterWebView.m#L451.

1.The current implementation uses the global `WKWebsiteDataStore.defaultDataStore` rather than the one used by `WebViewConfiguration`. However, the default `WKWebViewConfiguration.webSiteDataStore` == `WKWebsiteDataStore.defaultDataStore` according to documentation. So this code will function the same as the current implementation and will still be correct with future support for clearing caches for a single `WebView`. 

2. The current implementation clears all data types and not just cache datatypes.

Reply:
> 2\. The current implementation clears all data types and not just cache datatypes.

The method is just badly named; clearing everything is the documented behavior:

https://github.com/flutter/plugins/blob/main/packages/webview_flutter/webview_flutter/lib/src/webview.dart#L638-L643

While looking into this to see what the history of that code was, I saw that the Android implementation originally called `WebStorage.getInstance().deleteAllData();`, and your rewrite doesn't, so you actually regressed the Android behavior. Could you do a quick PR to fix that?

Reply:
> Could you do a quick PR to fix that?

Done: https://github.com/flutter/plugins/pull/5086